jedi/inference/gradual/type_var.py

```diff
@@ -108,8 +108,24 @@ def execute_annotation(self):
         return self._get_classes().execute_annotation()
 
     def infer_type_vars(self, value_set):
+        def iterate():
+            for v in value_set:
+                cls = v.py__class__()
+                if v.is_function() or v.is_class():
+                    cls = TypeWrapper(cls, v)
+                yield cls
+
         annotation_name = self.py__name__()
-        return {annotation_name: value_set.py__class__()}
+        return {annotation_name: ValueSet(iterate())}
 
     def __repr__(self):
         return '<%s: %s>' % (self.__class__.__name__, self.py__name__())
+
+
+class TypeWrapper(ValueWrapper):
+    def __init__(self, wrapped_value, original_value):
+        super().__init__(wrapped_value)
```

Comment:
Since this code is coming from me it's my bad, but the CI is failing. It should be `super(TypeWrapper, self)`.

We will finally get rid off Python 2 in the next few weeks, but until that point we still have to make it work :).

Reply:
D'oh. Of course. 7e637c5e